### PR TITLE
PR for Smiles2vec model porting

### PR DESCRIPTION
**Pull request by harishwar017**
## Description
First PR for porting Smiles2vec (nn.module)
Fix #(issue)

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [ ] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [x] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [x] Run `mypy -p deepchem` and check no errors
  - [x] Run `flake8 <modified file> --count` and check no errors
  - [x] Run `python -m doctest <modified file>` and check no errors
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by harishwar017** [comment]
@rbharath Can you please review this code. CI requires your approval, please look into it. Thanks!

File: deepchem/models/torch_models/smiles2vec.py [hunk 1]
```diff
@@ -0,0 +1,176 @@
+import torch.nn as nn
+import math
+from typing import List, Dict, Optional
+
+
+class Smiles2Vec(nn.Module):
```
Comment (rbharath) [inline deepchem/models/torch_models/smiles2vec.py:6]: Add to the rst docs

**Review by rbharath — COMMENTED** [review]
LGTM once CI is clear and doc fixes are in

File: deepchem/models/torch_models/smiles2vec.py [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+    """
+    Implements the Smiles2Vec model, that learns neural representations of SMILES
+    strings which can be used for downstream tasks.
+
+    The model is based on the description in Goh et al., "SMILES2vec: An
+    Interpretable General-Purpose Deep Neural Network for Predicting Chemical
+    Properties" (https://arxiv.org/pdf/1712.02034.pdf). The goal here is to take
```
Comment (GreatRSingh) [inline deepchem/models/torch_models/smiles2vec.py:13]: @harishwar017 We always keep the references in the end of the doc strings.

References
----------------
..[1] SMILES2vec: An
    Interpretable General-Purpose Deep Neural Network for Predicting Chemical
    Properties" (https://arxiv.org/pdf/1712.02034.pdf)
    
In this way, pls add these changes to it.

**Comment by GreatRSingh** [comment]
LGTM

cc: @rbharath @shreyasvinaya

**Pull request closed without merging**